app/Jobs/EDocument/MergeEDocument.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace App\Jobs\EDocument;
+
+use App\Models\Invoice;
+use App\Models\PurchaseOrder;
+use horstoeko\zugferd\ZugferdDocumentPdfBuilder;
+use Illuminate\Bus\Queueable;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+
+class MergeEDocument implements ShouldQueue
+{
+    use Dispatchable;
+    use InteractsWithQueue;
+    use Queueable;
+    use SerializesModels;
+
+    public $deleteWhenMissingModels = true;
+
+    public function __construct(private object $document, private string $pdf_file)
```

Comment:
Not sure you can use object as type here, I would suggest mixed, or using Union types for all possible documents.

ie. Invoice|Quote etc